src/deltacode/utils.py

```diff
@@ -179,39 +201,39 @@ def deltas(deltacode, all_delta_types=False):
     """
     for delta in deltacode.deltas:
         if all_delta_types is True:
-            yield delta.to_dict()
-        elif not delta.is_unmodified():
-            yield delta.to_dict()
+            yield delta.to_dict(deltacode)
+        elif not delta.status == "unmodified":
+            yield delta.to_dict(deltacode)
+
 
 def calculate_percent(value, total):
     """
     Return the rounded value percentage of total.
     """
-    ratio = (value / total) * 100
-    return round(ratio, 2)
+    try:
+        ratio = (value / total) * 100
+        return round(ratio, 2)
+    except ZeroDivisionError:
+        return 0
+
 
 class AlignmentException(Exception):
     """
     Named exception for alignment errors.
     """
+
     pass
 
-def align_trees(a_files, b_files):
-    """
-    Given two sequences of File objects 'a' and 'b', return a tuple of
-    two integers that represent the number path segments to remove
-    respectively from a File path in 'a' or a File path in 'b' to obtain the
-    equal paths for two files that are the same in 'a' and 'b'.
-    """
-    # we need to find one uniquly named file that exists in 'a' and 'b'.
+
+def align_trees(codebase1, codebase2):
     a_names = defaultdict(list)
-    for a_file in a_files:
-        a_names[a_file.name].append(a_file)
+    for resource in codebase1.walk():
```

Comment:
Please re-add the docstring to this function and update it to describe what is going on in this function.

Reply:
Added comments and doc strings at all required areas